src/Raven.Server/Web/System/DatabasesHandler.cs

```diff
@@ -83,6 +87,74 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
             }
         }
 
+        [RavenAction("/topology/modify", "POST", AuthorizationStatus.Operator)]
+        public async Task ModifyTopology()
+        {
+            var dbName = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name").Trim();
+            var raftRequestId = GetRaftRequestIdFromQuery();
+
+            await ServerStore.EnsureNotPassiveAsync();
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var json = await context.ReadForDiskAsync(RequestBodyStream(), "Database Topology");
+                var databaseTopology = JsonDeserializationCluster.DatabaseTopology(json);
+
+                if (LoggingSource.AuditLog.IsInfoEnabled)
+                {
+                    var clientCert = GetCurrentCertificate();
+
+                    var auditLog = LoggingSource.AuditLog.GetLogger("DbMgmt", "Audit");
+                    auditLog.Info($"Database \'{dbName}\' record has been modified by {clientCert?.Subject} ({clientCert?.Thumbprint})");
+                }
+
+                // Validate Database Name
+                DatabaseRecord databaseRecord;
+                using (context.OpenReadTransaction())
+                    databaseRecord = ServerStore.Cluster.ReadDatabase(context, dbName, out var index);
+
+                if (databaseRecord == null)
+                {
+                    throw new DatabaseDoesNotExistException($"Database {dbName} Record not found when trying to modify database topology");
```

Comment:
there is a static method called `Throw` or 'ThrowWithMessage' in that exception, please use it.

Reply:
Done